docs/en/manuals/extensions_best_practices.md

```diff
@@ -0,0 +1,131 @@
+# Best Practices
+
+Writing cross platform code can be difficult, but there are some ways to make it easier to both develop and maintain.
+Here we list some ways we at Defold work with cross platform native code and API's.
+
+## Defold code
+
+In the Defold engine we use C++ very sparingly. In fact, most code is very C-like.
+We avoid templates, except for a few container classes, due to the fact that templates both incurs a cost on compilation times
+as well as executable size.
+
+### C++ version
+
+The defold source it built with the default C++ version of each compiler (See [Native Extensions - Best Practices](/manuals/extensions_best_practices#_)).
+
+We avoid using the latest features or versions of C++. Mostly because we already have what we need to build an game engine.
+Keeping track of the latest features of C++ is a time consuming task, and to really master those features will require a lot of precious time.
+
+It also has the added benefit for our extension developers that we keep a stable ABI.
+
+### Standard Template Libraries - STL
+
+Since the Defold engine doesn't use any STL code, except for some algorithms and math (std::sort, std::upper_bound etc), it may work for you to use STL in your extension.
+
+Again, bear in mind that ABI incompatibilites may hinder you when using your extension in conjunction with other extensions or 3rd party libraries.
+
+Avoiding the (heavily tepmlated) STL libraries, also improves on our build times, and more importantly, the executable size.
+
+#### Strings
+
+In the Defold engine, we use `const char*` instead of `std::string`.
+
+`std::string` is a common pitfall when mixing different versions of C++ or compiler versions: you'll get an ABI mismatch.
+For us, it's better to use `const char*` and a few helper functions.
+
+### Make functions hidden
+
+Use the `static` keywork on functions local to your compile unit if possible. This lets the compiler do some optimizations, and
+can both improve performance as well as reduce executable size.
+
+# 3rd party libraries
+
+When choosing a 3rd party library to use (regardless of language), we consider at least these things:
+
+* Functionality - Does it solve the particular problem you have?
+* Performance - Does it infer a performance cost in the runtime? For each frame, or
+* Library size - How much bigger will the final executable be? Is it acceptable
+* Dependencies - Does it require extra libraries
+* Support - What state is the library in? Does it have many open issues? Is it still maintained? etc...
+* License - Is it ok to use for this project?
+
+
+# Open source dependencies
+
+Always make sure that you have access to your dependencies.
+E.g. if you depend on something on github, there's nothing preventing that repository either being removed, or suddenly changes
+direction or ownership.
```

Comment:
Perhaps suggest to fork it to maintain stability? Or not depending on latest master.